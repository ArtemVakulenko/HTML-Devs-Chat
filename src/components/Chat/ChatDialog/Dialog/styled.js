import styled from 'styled-components';

export const StDialogContainer = styled.div`
  height: 75%;
  overflow-x: hidden ;
  overflow-y: auto;
  &::-webkit-scrollbar {
    width: 5px;
    background-color: rgba(155, 155, 158, 0.15);
    border-radius: 5px;
    height: 12px;
  }

  &::-webkit-scrollbar-thumb {
    background: rgba(201, 201, 201, 0.5);
    border-radius: 5px;
  }
`;
