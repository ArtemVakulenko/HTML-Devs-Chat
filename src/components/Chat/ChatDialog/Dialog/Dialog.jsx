import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { useTheme } from 'src/components/hooks';
import { StDialogContainer } from './styled';
import MessageItem from '../MessageItem';

const Dialog = ({ messages, getMessages, match, userId }) => {
  useEffect(() => {
    const roomId = match.params.id;
    getMessages({ roomId, userId });
    messagesEndRef.current.scrollIntoView();
  }, [match]);
  useEffect(() => {
    messagesEndRef.current.scrollIntoView();
  }, [messages]);
  const messagesEndRef = useRef(null);
  const { colors, theme } = useTheme();

  return (
    <StDialogContainer colors={colors} theme={theme}>
      {messages.map((el) => (
        <MessageItem {...el} key={el.date} />
      ))}
      <div ref={messagesEndRef} />
    </StDialogContainer>
  );
};

Dialog.propTypes = {
  match: PropTypes.object,
  userId: PropTypes.string,
  activeRoom: PropTypes.object,
  getMessages: PropTypes.func,
  lastMessage: PropTypes.object,
  messages: PropTypes.arrayOf(PropTypes.object),
};

export default Dialog;
