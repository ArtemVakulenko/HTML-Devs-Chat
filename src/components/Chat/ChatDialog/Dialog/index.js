import { getMessages } from 'store/mainPage/actions';
import { messages, userId } from 'store/mainPage/selectors';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { compose } from 'redux';
import Dialog from './Dialog';

const mapStateToProps = (state) => ({
  messages: messages(state),
  userId: userId(state),
});
const mapDispatchToProps = (dispatch) => ({
  getMessages: (payload) => dispatch(getMessages(payload)),
});

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withRouter,
)(Dialog);
