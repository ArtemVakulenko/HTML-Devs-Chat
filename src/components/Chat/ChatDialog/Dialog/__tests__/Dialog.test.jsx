import React from 'react';
import configureStore from 'redux-mock-store';
import Dialog from '../Dialog';
import { shallowSmart } from '../../../../../__tests__/testHelper';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
const mockStore = configureStore();
const store = mockStore({
  sidebar: {
    search: '',
    rooms: [],
  },
  chat: {
    messages: [],
    roomId: '',
    messageValue: '',
    activeRoom: {},
  },
  user: {
    id: '',
    login: '',
    name: '',
    lastName: '',
    img: '',
    hobbie: '',
    city: '',
    age: '',
  },
});

jest.mock('react-notifications', () => ({
  NotificationManager: {
    error: jest.fn(),
    success: jest.fn(),
  },
}));
describe('Dialog', () => {
  describe('123', () => {
    const props = {
      messages: [],
      getMessages: jest.fn(),
      match: {},
      userId: '',
    };
    it('Should match snapshot', () => {
      const component = shallowSmart(<Dialog {...props} />, store);
      expect(component.html()).toMatchSnapshot();
    });
  });
});
