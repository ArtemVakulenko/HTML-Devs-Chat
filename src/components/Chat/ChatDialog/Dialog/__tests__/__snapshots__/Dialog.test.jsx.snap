// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog 123 Should match snapshot 1`] = `"<div class=\\"styled__StDialogContainer-hphmKj kldXao\\"><div></div></div>"`;

exports[`LoginPage 123 Should match snapshot 1`] = `"<div class=\\"styled__StDialogContainer-hphmKj kldXao\\"><div></div></div>"`;
